File: Cpp/7 kyu Get the Middle Character/getMiddle.cpp
/*
You are going to be given a word. Your job is to return the middle character
of the word. If the word's length is odd, return the middle character.
If the word's length is even, return the middle 2 characters.

Examples:
getMiddle("test") should return "es"
getMiddle("testing") should return "t"
getMiddle("middle") should return "dd"
getMiddle("A") should return "A"
*/

// Solution

#include <string>

std::string getMiddle(const std::string& s) {
  if (s.length() % 2 == 0) {
    return s.substr(s.length() / 2 - 1, 2);
  } else {
    return s.substr(s.length() / 2, 1);
  }
}

File: Rust/7 kyu Get the Middle Character/GetMiddle.rs
// You are going to be given a word. Your job is to return the middle character
// of the word. If the word's length is odd, return the middle character.
// If the word's length is even, return the middle 2 characters.

// Examples:
// get_middle("test") should return "es"
// get_middle("testing") should return "t"
// get_middle("middle") should return "dd"
// get_middle("A") should return "A"

// Solution

fn get_middle(s: &str) -> &str {
    let bytes = s.as_bytes();
    let mid = bytes.len() / 2;
    if bytes.len() % 2 == 0 {
        &s[mid - 1..mid + 1]
    } else {
        &s[mid..mid + 1]
    }
}
